tools/sourcemap-tools/src/SourceProcessor.ts

```diff
@@ -0,0 +1,71 @@
+import fs from 'fs';
+import { BasicSourceMapConsumer, Position, RawSourceMap, SourceMapConsumer, SourceMapGenerator } from 'source-map';
+import { DebugIdGenerator } from './DebugIdGenerator';
+import { stringToUuid } from './helpers/stringToUuid';
+
+export class SourceProcessor {
+    constructor(private readonly _debugIdGenerator: DebugIdGenerator) {}
+
+    public async processSourceAndSourceMap(source: string, sourceMap: string | RawSourceMap, debugId?: string) {
+        if (!debugId) {
+            debugId = stringToUuid(source);
```

Comment:
Let's pretend I have a colossal javascript file—1000 lines of code. I'm changing the last one.  If I uploaded source maps before the change, will this generate the same uuid after my change?

Reply:
ah ok got it. I missed that part.